### ISSUE TITLE
Set up integration test with `libprio-cc`

### ISSUE THREAD

**Issue by tgeoghegan** [abetterinternet/libprio-rs#71]
Our distinguished colleagues at Google have written a [Prio implementation designed to be compatible with this implementation](https://github.com/google/libprio-cc) (hi @tlepoint!). As we add more features to `libprio-rs` to support PPM, we want to make sure we don't break ENPA as we introduce new encodings or API surfaces. A good way to achieve this would be to set up an integration test where we use `libprio-cc` to construct inputs and `libprio-rs` to aggregate them.

**Comment by cjpatton** [comment]
Good idea. FWIW there are some test vectors here that should ensure compatibility for the current parameters. Don't know what Google's plans are for their library though.

**Comment by tgeoghegan** [comment]
I spent a little time trying to get `libprio-cc` to compile yesterday, and I concluded that it' nots worthwhile to figure out how to build `libprio-cc` in `libprio-rs`, since we'd need a C++ toolchain and also apparently an Android SDK. This would already be a lot of work to set up in GitHub actions, and a huge pain in the butt for developers to run locally.

Instead, I want to test backward compatibility with ENPA (which is what testing against `libprio-cc` was really meant to achieve) by testing against some test vectors, recorded with a version of `libprio-rs` known empirically to work in `prio-server` and hence ENPA. Pull request #88 achieves this with a `FieldPriov2` based test vector.

Once we take that change, I think we should release `prio` 0.5.0 to crates.io, and then empirically verify that a `prio-server` built against `prio` 0.5.0 and using `FieldPriov2` can still perform aggregations with a `prio-server` using `prio` 0.4.0. If that works, then we will know that 0.5.0 is compatible with ENPA, and thus that test vectors generated from 0.5.0 can be used to maintain backward compatibility going forward.

The other "obvious" thing to do would be to generate test vectors from the already shipped `prio` 0.4.0 which we know to work just fine in `prio-server`. However that version depends on some crypto crates that have been yanked from crates.io, and so building it is surprisingly difficult!

**Comment by cjpatton** [comment]
SGTM. Once we've generated the test vectors we should send a PR to `libprio-cc` that runs them against their server.
